### PR TITLE
Add ktlint to Maven build

### LINKED ISSUE #255: Integrate ktlint with Maven build lifecycle
I spotted that we're using ktlint for linting of our Kotlin code. It's currently executed as part of the CI build via a GitHub action but I wondered whether it's worth integrating this with the Maven build as per their [docs](https://github.com/pinterest/ktlint#-with-maven) or maybe via this dedicated [Maven plugin](https://github.com/gantsign/ktlint-maven-plugin)?

The main benefit here is tying the linting into the standard Maven build life-cycle so we get faster and simpler (i.e. local setup) feedback rather than finding out we've broken a linting rule after pushing and waiting for CI to complete.

### PR DESCRIPTION
**Pull request by nickebbitt**
Linting rules will execute automatically as part of the `validate` phase of the Maven build life-cycle.

Also remove from GitHub action workflow to avoid duplication of work.

Fixes #255 

# Checklist

- [ ] `mvn clean install` build and test completes
- [ ] You changed or added to the documentation

### CODE REVIEW

**Comment by nickebbitt** [comment]
I think the codecov bot has failed due to this PR applying the linting rules to many files which haven't functionally changed.

File: adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/OpenApiDocs.kt [hunk 1]
```diff
@@ -2,7 +2,8 @@ package net.adoptopenjdk.api.v3
 
 object OpenApiDocs {
 
-    const val RELASE_NAME = """
+    const val RELASE_NAME =
```
Comment (karianna) [inline adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/OpenApiDocs.kt:5]: This seems like an odd klint rule - do we really want this?

Comment (nickebbitt) [reply adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/OpenApiDocs.kt:5]: Maybe not, and actually that reminds me, I noticed a number of rules being disabled in-line in the code. We could probably now just pull them, with any others we don't want into a file where we can control the rules?

For this PR, we could always just move the linting to Maven but stay on the same version and therefore the same default ruleset?

Comment (nickebbitt) [reply adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/OpenApiDocs.kt:5]: For this rule, I actually think the alignment of the opening and closing `"""` is useful.

Comment (karianna) [reply adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/OpenApiDocs.kt:5]: I think to check with @johnoliver  on the disabled rules - I'd prefer to see them disabled in klint config as opposed to in code, but would be good to check.

Comment (nickebbitt) [reply adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/OpenApiDocs.kt:5]: It seems that ktlint is very opinionated and I don't think there's actually support for a config file with custom rule-sets - see https://ktlint.github.io/#faq-error-suppression

This is probably why @johnoliver has suppressed them in-line where needed.

My thoughts are if we are using `ktlint` then maybe we just adopt its style completely (which includes [no wildcard imports](https://ktlint.github.io/#rule-import) btw), except for when there is a really strong reason not to. In these exceptional cases, which should be afew and far between, an in-line suppression may be ok.

If we do want more flexibility and custom rule-sets then I can recommend [detekt](https://github.com/detekt/detekt).

**Comment by johnoliver** [comment]
Since this will likely kill all other WIP with merge conflicts I will hold off merging this until we can get other PRs in and sync staging and master

**Comment by nickebbitt** [comment]
Hi @johnoliver - is this one safe to merge now?

**Comment by nickebbitt** [comment]
Following the master/sating sync and subsequent merge I'm not happy with how this looks. I'm going to recreate the PR so it's a cleaner diff/merge

**Pull request closed without merging**